Tests/NIOHTTP2Tests/SimpleClientServerInlineStreamMultiplexerTests.swift

```diff
@@ -329,12 +329,55 @@ class SimpleClientServerInlineStreamMultiplexerTests: XCTestCase {
         stream.writeAndFlush(headerPayload, promise: promise)
     }
 
-    func testSuccessfullyReceiveAndSendPingEvenWhenConnectionIsFullyQuiesced() throws {
+    func testOpenStreamBeforeReceivingAlreadySentGoAway() throws {
         let serverHandler = InboundFramePayloadRecorder()
         try self.basicHTTP2Connection() { channel in
             return channel.pipeline.addHandler(serverHandler)
         }
 
+        let clientHandler = InboundFramePayloadRecorder()
+        let childChannelPromise = self.clientChannel.eventLoop.makePromise(of: Channel.self)
+        let multiplexer = try self.clientChannel.pipeline.handler(type: NIOHTTP2Handler.self).wait().multiplexer.wait()
+        multiplexer.createStreamChannel(promise: childChannelPromise) { channel in
+            return channel.pipeline.addHandler(clientHandler)
+        }
+        self.clientChannel.embeddedEventLoop.run()
+        let childChannel = try childChannelPromise.futureResult.wait()
+
+        // Server sends GOAWAY frame.
+        let goAwayFrame = HTTP2Frame(streamID: .rootStream, payload: .goAway(lastStreamID: .maxID, errorCode: .noError, opaqueData: nil))
+        serverChannel.writeAndFlush(goAwayFrame, promise: nil)
+
+        // Client sends headers.
+        let headers = HPACKHeaders([(":path", "/"), (":method", "POST"), (":scheme", "https"), (":authority", "localhost")])
+        let reqFramePayload = HTTP2Frame.FramePayload.headers(.init(headers: headers))
+        childChannel.writeAndFlush(reqFramePayload, promise: nil)
+
+        self.interactInMemory(self.clientChannel, self.serverChannel, expectError: true) { error in
+            XCTAssert(error is NIOHTTP2Errors.IOOnClosedConnection)
```

Comment:
The original problem was that the race between the GOAWAY and HEADERS resulted in a connection error on the server. 

While we're now sending a RST_STREAM frame to the client, we're still propagating a connection-level error on the server. 

I think this the server propagate a `StreamError` which uses `CreatedStreamAfterGoaway` as its `baseError` and we should be catching that here.

Reply:
Oh yeah, that makes sense. I didn't want to assert that a GOAWAY had been sent, due to this statement: "[A GOAWAY frame might not immediately precede closing of the connection](https://www.rfc-editor.org/rfc/rfc9113.html#section-6.8-7)". But I can see that I mistook a fully quiesced connection for a closed one.